File: src/main/java/com/github/zametki/component/form/RefreshCaptchaLink.java
package com.github.zametki.component.form;

import org.apache.wicket.ajax.AjaxRequestTarget;
import org.apache.wicket.ajax.markup.html.AjaxLink;
import org.apache.wicket.markup.html.image.Image;

public class RefreshCaptchaLink extends AjaxLink<Void> {
    private final CaptchaField captchaField;
    private final Image captchaImage;

    public RefreshCaptchaLink(String id, CaptchaField captchaField, Image captchaImage) {
        super(id);
        this.captchaField = captchaField;
        this.captchaImage = captchaImage;
    }

    public void onClick(AjaxRequestTarget target) {
        captchaField.change();
        captchaField.invalidate();
        target.add(captchaImage);
        target.add(captchaField);
    }
}

File: src/main/java/com/github/zametki/component/form/RefreshCaptchaLink.kt
package com.github.zametki.component.form

import org.apache.wicket.ajax.AjaxRequestTarget
import org.apache.wicket.ajax.markup.html.AjaxLink
import org.apache.wicket.markup.html.image.Image

class RefreshCaptchaLink(id: String, private val captchaField: CaptchaField, private val captchaImage: Image) : AjaxLink<Void>(id) {

    override fun onClick(target: AjaxRequestTarget) {
        captchaField.change()
        captchaField.invalidate()
        target.add(captchaImage)
        target.add(captchaField)
    }
}
